### PR TITLE
Remote structure

### PR DESCRIPTION
**Pull request by jawoznia**
PR1 of #28

### CODE REVIEW

File: sylvia/tests/remote.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    impl SomeInterface for super::SomeContract {
+        type Error = StdError;
+    }
+}
+
+#[contract]
+#[messages(some_interface as SomeInterface)]
+impl SomeContract {
+    #[allow(clippy::new_without_default)]
```
Comment (hashedone) [inline sylvia/tests/remote.rs:23]: Naaaaaaaaaaaaaaaaaaaaaaaaaaaaaaah, this is ZST. Don't even implement new on it, you just create it in place.

File: sylvia/tests/remote.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+use cosmwasm_std::{Response, StdResult};
+use sylvia::{contract, types::InstantiateCtx};
+
+pub struct SomeContract {}
```
Comment (hashedone) [inline sylvia/tests/remote.rs:4]: You ain't need `{}` here. ZST you are doing just `pub struct SomeContract;`.

File: sylvia/tests/remote.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use cosmwasm_std::Addr;
+
+    use crate::some_interface;
+
+    #[test]
+    fn remote_generation() {
```
Comment (hashedone) [inline sylvia/tests/remote.rs:41]: Good.

**Review by hashedone — COMMENTED** [review]
Overall good. The utility for accessing `Addr` remotely is missing. Additionally, minor fixes in tests.

File: sylvia/tests/remote.rs (hunk 1 above)
Comment (jawoznia) [reply sylvia/tests/remote.rs:23]: We require the `new` method in both entry_points and multitests generation. It is currently obligatory to implement it on contract.

Comment (hashedone) [reply sylvia/tests/remote.rs:23]: Ok, makes sense - still, remove `{}`, and just derive default as it is oneliner.

Comment (jawoznia) [reply sylvia/tests/remote.rs:23]: We could check if there is a new method and otherwise create the contract like unit type.

Comment (jawoznia) [reply sylvia/tests/remote.rs:23]: or require the default trait implementation on it

File: sylvia-derive/src/input.rs [hunk 4]
```diff
@@ -62,12 +64,17 @@ impl<'a> TraitInput<'a> {
                     #messages
 
                     #multitest_helpers
+
+                    #remote
```
[the remark this answers is not in the document]
Comment (jawoznia) [reply sylvia-derive/src/input.rs:69]: Seems skipping tarpaulin didn't do the job

[the remark this answers is not in the document]
Comment (jawoznia) [reply sylvia-derive/src/input.rs]: Wrapping quote in scope did the job although I don't really like this extra nesting

File: sylvia/tests/remote.rs (hunk 1 above)
Comment (hashedone) [reply sylvia/tests/remote.rs:23]: No, don't diverge basing on existence of `new()`. This is a bookcase for Default trait

**Review by hashedone — APPROVED** [review]
LGTM

**Pull request merged**